lib/translator/inquiry.cc

```diff
@@ -13,10 +13,104 @@
+// anonymous namespace for helper functions and constant expressions
+namespace {
+
+// Builds the IdentificationDescriptor according to 6.1.4.5 of the translation
+// spec:
+// https://nvmexpress.org/wp-content/uploads/NVM_Express_-_SCSI_Translation_Reference-1_5_20150624_Gold.pdf
+// This only builds the constant part of the Identification Descriptor. The
+// caller is responsible for building the identifier.
+scsi_defs::IdentificationDescriptor BuildIdentificationDescriptor(
+    const nvme_defs::IdentifyNamespace &identify_namespace_data) {
+  // check if nonzero
+  bool nguid_nz =
+      identify_namespace_data.nguid[0] || identify_namespace_data.nguid[1];
+  bool eui64_nz = identify_namespace_data.eui64;
+  uint8_t identifier_length;
+
+  if (nguid_nz) {
+    // Shall be set to 08h if the EUI-64 field is populated from the NVMe EUI64
+    // field.
+    identifier_length = kIdentifierLengthNguid;
+
+    uint8_t identification_descriptor_identifier[16];
+
+    memcpy(&identification_descriptor_identifier,
+           &identify_namespace_data.nguid,
+           sizeof(identification_descriptor_identifier));
+
+  } else if (eui64_nz) {
+    // Shall be set to 10h if the EUI-64 field is populated from the NVMe NGUID
+    // field.
+    identifier_length = kIdentifierLengthEui64;
+
+    uint64_t identification_descriptor_identifier =
+        identify_namespace_data.eui64;
+  } else {
+    // function not supported as both NGUID and EUI64 fields are zero
+    DebugLog("Both NGUID and EUI-64 fields are zero in IdentityNamespaceData");
+  }
+
+  scsi_defs::IdentificationDescriptor identification_descriptor = {
+
+      // Shall be set to 0h. PIV field shall indicate this field is reserved as
+      // no
+      // specific protocol to be identified. This field will be ignored!
+      .protocol_identifier = scsi_defs::ProtocolIdentifier::kFibreChannel,
+
+      // Shall be set to 1h indicating associated fields are in binary format.
+      .code_set = scsi_defs::CodeSet::kBinary,
+
+      // Shall be set to 0b indicating PROTOCOL IDENTIFIER field is reserved.
+      .protocol_identifier_valid = 0,
+
+      // Shall be set to 00b indicating DESIGNATOR field is associated with
+      // logical unit.
+      .association = scsi_defs::Association::kPhysicalDevice,
+
+      // Shall be set to 2h indicating EUI-64 based identifier.
+      .identifier_type = scsi_defs::IdentifierType::kEUI64,
+
+      .identifier_length = identifier_length};
+  return identification_descriptor;
+}
+}  // namespace
+
+// Creates and validates a Inquiry Command struct
+StatusCode RawToScsiCommand(absl::Span<const uint8_t> raw_cmd,
+                            scsi_defs::InquiryCommand &cmd) {
+  if (raw_cmd.empty()) {
+    DebugLog("buffer is empty or nullptr");
+    return StatusCode::kInvalidInput;
+  }
+
+  uint8_t raw_opcode = raw_cmd[0];
+  if (raw_opcode > 0xaf) {
+    DebugLog("raw opcode is out of range: %x", raw_opcode);
+    return StatusCode::kInvalidInput;
+  }
```

Comment:
I'm not sure if this is just old code as a result of not merging in the base branch changes, but these checks (including whether the opcode == kInquiry) is no longer needed because it is covered by  Translation::Begin

Reply:
it is unused, @basimsahaf check out `standard-inquiry` for updated inquiry code integrated w the generic translator